File: models/light_v0.cpp
#define _USE_MATH_DEFINES  // Required to use M_PI with MSVC
#include <math.h>

/*!
  @brief Compute the intensity of light.

  @param[in] doy Day of year.
  @param[in] height Distance from ground in cm.

  @returns intensity Intensity of light in ergs cm^-2 s^-1.
 */
double light(double doy, double height) {
  // Define parameters that are static across a run
  double amplitude = 80.0;
  double doy_offset = 0.0;

  // Calculate intensity
  double intensity = amplitude * height * (1.0 + sin(2.0 * M_PI * (doy - doy_offset) / 365));
  
  return intensity;
}

File: models/light_v0.f90
!---------------------------------------------------------------------------
!> @brief Compute the intensity of light.
!
!> @param[in] doy: Day of year.
!> @param[in] height: Distance from ground in cm.
!
!> @return intensity: Intensity of light in ergs cm^-2 s^-1.
!---------------------------------------------------------------------------
function light(doy, height) result(intensity)
  real(kind=8) :: doy
  real(kind=8) :: height
  real(kind=8) :: intensity
  real, parameter :: Pi = 3.1415927
  
  ! Define parameters that are static across a run
  real, parameter :: amplitude = 80.0
  real, parameter :: doy_offset = 0.0

  ! Calculate intensity
  intensity = amplitude * height * (1.0 + SIN(2.0 * Pi * (doy - doy_offset) / 365))
end function light
